### PR TITLE
WIP: moving contents to wiki

### LINKED ISSUE #883: rm deprecated sections from README
Remove sections from code.openlibrary.org and README and shift it to the Wiki.

### PR DESCRIPTION
**Pull request by mekarpeles**
Closes #883

## Description:
- moves `testing` out of README and out of `olsystem` and into wiki
- moves "join slack" to README

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ catalog, building towards a web page for every book ever published.
      - [The Backend](#the-backend)
      - [The Service Architecture](#the-service-architecture)
    - [Developer's Guide](#developers-guide)
-   - [Running Tests](#running-tests)
+   - [Running Tests](https://github.com/internetarchive/openlibrary/wiki/Testing)
    - [Contributing](#contributing)
    - [Public APIs](https://openlibrary.org/developers/api)
    - [FAQs](https://openlibrary.org/help/faq)
```
Comment (salman-bhai) [inline Readme.md]: Can we move FAQs as well to the Wiki? It'll be much better over there!

Comment (mekarpeles) [reply Readme.md]: sounds good to me

File: Readme.md [hunk 2]
```diff
@@ -104,23 +98,15 @@ how to use Infogami (and its database, Infobase):
...
 - [Overview of OpenLibrary Service Architecture](https://openlibrary.org/about/architecture)
 
-## Running tests
-
-Open Library tests can be run using pytest (py.test).
-
-Inside vagrant, go to the application base directory:
-
-        cd /openlibrary
-        make test
```
Comment (hornc) [inline Readme.md]: Why are these removed? I think testing instructions belong in the repo's readme (and are important!) These instructions are what I use to test, so I consider them accurate. If they change, they will change with the code, so should be version controlled with it too.

File: Readme.md (hunk 1 above)
Comment (hornc) [reply Readme.md]: The FAQ is part of the live site, I don't think a dev wiki is an appropriate location. If the intent is to move out of date content out of the repo, perhaps have a stripped down placeholder to indicate the live site expects a page at that location, and provide a link to the live page to aid developers, but make it clear the FAQ content is stored in the production db.

File: Readme.md (hunk 2 above)
Comment (tfmorris) [reply Readme.md]: I agree that anything dev-centric should have its primary home in git.

File: Readme.md [hunk 3]
```diff
@@ -104,23 +98,15 @@ how to use Infogami (and its database, Infobase):
...
-
-### Integration Tests
```
Comment (salman-bhai) [inline Readme.md]: I think we can probably add these back. Or maybe have a link in the Wiki given that we do have some documentation regarding Testing in the Wiki.

File: Readme.md (hunk 2 above)
Comment (salman-bhai) [reply Readme.md]: Maybe we could link everything to this page and add the rest [here](https://github.com/internetarchive/openlibrary/wiki/Testing).
What do you think @tfmorris and @hornc
